### PR TITLE
Fix router dependencies

### PR DESCRIPTION
**Pull request by desertaxle**
#16524 was failing the linting checks in CI. Turns out we were using the wrong `zod` adapter. This PR installs the correct `zod` adapter, uninstalls the incorrect one, and adds `@tanstack/react-router` as an explicit dependency in `package.json`.

### CODE REVIEW

File: ui-v2/src/routes/runs/flow-run.$id.tsx [hunk 1]
```diff
@@ -1,16 +1,9 @@
...
 	component: RouteComponent,
 });
 
 function RouteComponent() {
-	return (
-		<div className="flex flex-col gap-4">
-			<DeploymentsPageHeader />
-			<DeploymentsEmptyState />
-		</div>
-	);
```
Comment (desertaxle) [inline ui-v2/src/routes/runs/flow-run.$id.tsx:15]: This ended up in the wrong route. It's probably my fault, so I thought I'd try and sneak a fix in here 😅

File: ui-v2/src/components/ui/flow-run-acitivity-bar-graph/flow-run-activity-bar-graph.test.tsx [hunk 2]
```diff
@@ -62,12 +67,24 @@ describe("FlowRunActivityBarChart", () => {
...
+			history: createMemoryHistory({
+				initialEntries: ["/"],
+			}),
+			context: {
+				queryClient: new QueryClient(),
+			},
+		});
+
+		// @ts-expect-error - Type error from using a test router
+		render(<RouterProvider router={router} />);
```
Comment (desertaxle) [inline ui-v2/src/components/ui/flow-run-acitivity-bar-graph/flow-run-activity-bar-graph.test.tsx:87]: This appears to be the 'best' way to test components with a router dependency based on the [`@tanstack/react-router` tests](https://github.com/TanStack/router/blob/main/packages/react-router/tests/routeContext.test.tsx).

**Comment by devinvillarosa** [comment]
Nice, I was wondering what was needed to fix that upgrade

File: ui-v2/src/components/ui/flow-run-acitivity-bar-graph/flow-run-activity-bar-graph.test.tsx (hunk 2 above)
Comment (devinvillarosa) [reply ui-v2/src/components/ui/flow-run-acitivity-bar-graph/flow-run-activity-bar-graph.test.tsx:87]: We can probably add this to [our test utils](https://github.com/PrefectHQ/prefect/blob/main/ui-v2/tests/utils/index.ts) to expedite the router wrapping for future tests

**Pull request merged**